docs/sources/flow/reference/components/loki.source.kafka.md

```diff
@@ -0,0 +1,139 @@
+`labels`                 | `map(string)`          | The labels to associate with each received Kafka event.  | `{}`         | no
+`forward_to`             | `list(LogsReceiver)`   | List of receivers to send log entries to.                |              | yes
+`relabel_rules`          | `RelabelRules`         | Relabeling rules to apply on log entries.                | `{}`         | no
+
+`assignor` values can be either `"range"`, `"roundrobin"`, or `"sticky"`.
+
+The `relabel_rules` field can make use of the `rules` export value from a
+`loki.relabel` component to apply one or more relabeling rules to log entries
+before they're forwarded to the list of receivers in `forward_to`.
+
+## Blocks
+
+The following blocks are supported inside the definition of `loki.source.kafka`:
+
+Hierarchy | Name | Description | Required
+--------- | ---- | ----------- | --------
+authentication | [authentication] | Optional authentication configuration with Kafka brokers. | no
+authentication > tls_config | [tls_config] | Optional authentication configuration with Kafka brokers. | no
+authentication > sasl_config | [sasl_config] | Optional authentication configuration with Kafka brokers. | no
+authentication > sasl_config > tls_config | [tls_config] | Optional authentication configuration with Kafka brokers. | no
+
+[authentication]: #authentication-block
+[tls_config]: #tls_config-block
+[sasl_config]: #sasl_config-block
+
+### authentication block
+
+The `authentication` block defines the authentication method when communicating with the Kafka event brokers.
+
+Name                     | Type          | Description | Default | Required
+------------------------ | ------------- | ----------- | ------- | --------
+`type`                   | `string`      | Type of authentication. | `"none"` | no
+
+`type` supports the values `"none"`, `"ssl"`, and `"sasl"`. If `"ssl"` is used,
+you must set the `tls_config` block. If `"sasl"` is used, you must set the `sasl_config` block.
+
+### tls_config block
+
+{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}
+
+### sasl_config block
+
+The `sasl_config` block defines the listen address and port where the listener
+expects Kafka messages to be sent to.
+
+Name                     | Type          | Description | Default | Required
+------------------------ | ------------- | ----------- | ------- | --------
+`mechanism` | `string` | Specifies the SASL mechanism the client uses to authenticate with the broker. | `"PLAIN""` | no
+`user`      | `string` | The user name to use for SASL authentication. | | no
+`password`  | `string` | The password to use for SASL authentication. | | no
+`use_tls`   | `bool`   | If true, SASL authentication is executed over TLS. | `false` | no
+
+## Labels
+
+The `labels` map is applied to every message that the component reads.
+
+The following internal labels prefixed with `__` are available but are discarded if not relabeled:
```

Comment:
Question: Is this component susceptible to the same issue as other components where no labels are populated by default, therefore by discarding these we'll get an empty label set and the message will never be sent to Loki?

Reply:
that sounds right to me, but I don't know 100% how it works

Reply:
I tested and yes, this is how it works